### ISSUE TITLE
Couldn't convert b64 to image from getText()

### ISSUE THREAD

**Issue by asshatter** [rk700/PyMuPDF#154]
I got Json output for page.getText() as : 
```
{u'blocks': [{u'bbox': [0, 0, 792, 612.48],
   u'height': 3300,
   u'image': u'///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////v//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////AGf/DgeAY/AAAAAYAAAAAAHAAf/+P///P/+AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAH////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////4AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////+AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////gAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////4AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////+AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////gAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////4AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////+AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////gAAAAAAAAAAAAAH8APnwDgc8AMAAAAAAAAAAAAAAAcAGAA5/P///////////////////////////////////AP//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////9x///////////////////////////4f//////////////////////////////////////////////////////////////////////////////////wD///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////f///////////////////////////+H//////////////////////////////////////////////////////////////////////////////////4A//////////////////////////////////////////////////////////////////////////////////////////////// (more image data)/'
.
.
.
```

Andi tried converting it to PIL Image by io as : 
```
image = json.loads(pg.getText('json'))['blocks'][0]['image']  
img = Image.open(io.BytesIO(base64.b64decode(image))) #THIS CODE WORKED FOR ALL CASES BUT FAILING FOR ONE CASE
```
And I got the following error : 
```
---------------------------------------------------------------------------
IOError                                   Traceback (most recent call last)
<ipython-input-15-37edbb51d0e1> in <module>()
----> 1 img = Image.open(io.BytesIO(base64.b64decode(image)))

/usr/local/lib/python2.7/site-packages/PIL/Image.pyc in open(fp, mode)
   2583         fp.close()
   2584     raise IOError("cannot identify image file %r"
-> 2585                   % (filename if filename else fp))
   2586 
   2587 #

IOError: cannot identify image file <_io.BytesIO object at 0x10c6e2830>
```

**Comment by JorjMcKie** [comment]
I just reconfirmed that the method itself does work. I successfully successfully converted a b64 image from getText() back to binary and created a Pixmap from it.
So something special must exist in your situation. Please answer the following questions:
* what is your PyMuPDF version?
* is this image extractable from the PDF (via fitz.Pixmap(doc, xref))?
* can you share the PDF if all else fails?

**Comment by JorjMcKie** [comment]
Another thing you can try is using "html" option of getText and save the result to a .html file. Try to open  this with your browser.

**Comment by asshatter** [comment]
Yep, the html version gives the correct b64 string.
```
<div style="width:792pt;height:612pt">
<img style="top:0pt;left:0pt;width:792pt;height:612pt" src="data:image/png;base64,\niVBORw0KGgoAAAANSUhEUgAACfgAAAzkCAAAAAAsQ+LGAAAAz2lDQ1BNdVBERiBQ\ncm9maWxlAAB4nGNgYFzAAARMAgwMuXklRe5BjpERkVEKDEggMbm4wDEgwIcBJ/h2\njYERRF/Wxa0GJ2BNSS1OBtIHgLg2uaCohIGB0QHI1igvKQCxM4BskaRsMLsGxM4O\nCXIGsicA2XxQvSCg7FhUkpmWWqEQnJ9WUp5YlKpQ7F6UWKng6eysEFCUn5aZk0qG\n6wiAktSKEhDtnF9QWZSZnlGigOEKIwNDQwZEmH4OAYcVo9h59HAGgeTSojIok5Hx\nLAMDALgON9FQDpxXAAGugUlEQVR4nOyd64KrqBJGk/d/aM+Z3UkEkeLjjrrWj5nd\n3RVARFgWJnlvLwDoxpsrDAAA1oFlCQAAAOAhIH4AAAAADwHxAwAAAHgIiB8AAADA\nQ0D8AAAAAB4C4gcAAADwEBA/AAAAgIeA+AEAAAA8BMQPAAAA4CEgfgAAAAAPAfED\nAAAAeAiIHwAAAMBDQPwAAAAAHgLiBwAAAPAQED8AAACAh4D4AQAAADwExA8AAADg\nISB+AAAAAA8B8QMAAAB4CIgfAAAAwENA/AAAAAAeAuIHAAAA8BAQPwAAAICHgPgB\nAAAAPATEDwAAAOAhIH4AAAAADwHxAwAAAHgIiB8AAADAQ0D8AAAAAB4C4gcAAADw\nEBA/AAAAgIeA+AEAAAA8BMQPAAAA4CEgfgAAAAAPAfEDAAAAeAiIHwAAAMBDQPwA\nAAAAHgLiBwAAAPAQED8AAACAh4D4AQAAADwExA8AAADgISB+AAAAAA8B8QMAAAB4\nCIgfAAAAwENA/AAAAAAeAuIHAAAA8BAQPwAAAICHgPgBAAAAPATEDwAAAOAhIH4A\nAAAADwHxAwAAAHgIiB8AAADAQ0D8AAAAAB4C4gcAAADwEBA/AAAAgIeA+AEAAAA8\nBMQPAAAA4CEgfgAAAAAPAfEDAAAAeAiIHwAAAMBDQPwAAAAAHgLiBwAAAPAQED8A\nAACAh4D4AQAAADwExA8AAADgISB+AAAAAA8B8QMAAAB4CIgfAAAAwENA/AAAAAAe\nAuIHAAAA8BAQPwAAAICHgPgBAAAAPATEDwAAAOAhIH635u3/yMkGAAB4NIhfI/5T\nrO0VqNZacLIBAAAezYLit7Q6XZv1TjYAAAAMBPF7EuudbAAAABgI4vck1jvZAAAA\nMBDE70msd7IBAABgIOuJH97Xj+VONgAAAIxkJfFD+Xqz0MkGAACA8SB+T2Khkw0A\nAADjQfyuz+b3XHhG3/E/AQAAwIO4tPj9M57fpyaLR/KN/X7isvvRy/8VuCXasVCH\nqSB+AAAA8B/Lid9C7bkNiB8AAAD8x2rit1Bz7gPiBwAAAP+B+D0AxA8AAAD+A/F7\nAIgfAAAA/Afi9wAQPwAAgEfw/r3xNR6wDPcUv/f8Y0L8AAAA4D9WE7/buckKNov4\nAQAAwH8gfp1Z4aAQPwAAAPgPxK8zKxwU4gcAAAD/gfh1ZoWDQvwAAACewOHLx77r\nvmN7iF9nVjgoxA8AAAD+A/HrDG/uAAAAgFVYTfwWak4bVjgoxA8AAAD+A/HrzAoH\nhfgBAADAf6wmfrdzE8QPAAAAVgHx6wziBwAAAKuA+HUG8QMAAIBVQPw6g/gBAADA\nKiB+nUH8AAAAYBUQv84gfgAAALAKiF9nED8AAABYBcSvM4gfAAAArALi1xnEDwAA\nAFYB8esM4gcAAACrgPh1BvEDAACAVUD8OoP4AQAAwCqsJn4LNacNK9gs4gcAAAD/\n8XTx614n4gcAAACrcDHx+1OYeNC3CDHua0QdOwHxAwCAZ/EuWG5WWC2fwWrid37W\nvx6XErW8OId+vbDCUEb8AABmsMIKkORnaZ+1wm6vu35aa6x81G8vD7N8b92AS4hf\n4GmRkVEe12+orXDZc0EBwIVIT5uKfGT6zB4oTJTJwCvJzM/S9l61t8tcjEVWO+hP\nqFPu0p11C64mft993FjY5lyQ0XuO4wh7I34AcDnkp2PUS3/2dKXaki4fms+oe0V5\nm0qXkRn/Eak/5O2yxDadVvunr8x1uzcnR7b0WavhCuLnPJC3fX8RET9/3MTjpPKa\nMHsm/bXhPxY62QAd8OYz7SmjoqvfKrpiOpHSWBmPKPtL2fLpKdmWRPmQfUbdK8oM\nnC4zqsrkLImHQ0mmV5KHPXQ9TjUk4KZr5kXETxW6nDjnzx1H2rTbl0Mb/jG7IbAa\neSNfncvT+pV7xeU8erQ5P4t7dtkXhlV08SWvpbEyHlFWA1dJT8m2JMqHrBTinpIe\nuIjMyCrzE//N+0WkyO3wi1gXbFLeY++r/stxsiEBN10zLyR++3hA/PLb8I/ZDYFm\n6BspyWLUQrx5MZllsvUr67IQ81bBfp4uftkdmRK/kjOTt4eWjssMnJ6eyvM0XT7S\nPpNbcU7gVJnpI37b8Rd2kZG/H0v0Kp944+Fz1yUT8XvFh24LSleB5m34j4VO9n9I\naRwvOB2amGhy5l93Glik57wh+4/KhmUMTzEfpelXzmWRlbdyIzLET4jMWUSlAs9e\nJ6SxguNMr9F/Ra2entJtKV8+kuInrRwlgXNlJuNmoo/4CTPVKn31WiNNM4YriV/8\nF96vFfHzft/1bD9X/BKrh5jG8W4co4GyEOnnWvScsTj3LF/q2qVLSriEJNNMxlWY\nIUc5ibDt+I+MA0uF5opf7plR01i5Wac9ZPH0lG4AncQvmVrIDpwvM+KE10/80pPV\nKsPPacv9uYj4He+EjRlxT/mk7nCd4hC/LrUa9YlpnE/3CXOHJkT................ 
```

I don't know why json is returning gibberish. Can you look into it.

**Comment by JorjMcKie** [comment]
Please send me the PDF.
Please also try my other suggestion: convert the bas64 string and then Try creating a Pixmap via fitz.Pixmap(bytearray(binary string)), or by fitz.Pixmap(doc, xref).

Gesendet von Mail<https://go.microsoft.com/fwlink/?LinkId=550986> für Windows 10

**Comment by asshatter** [comment]
When I tried fitz.Pixmap(doc, xref) for a page I got this : 
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-60-c8a9824a91b5> in <module>()
----> 1 fitz.Pixmap(doc, 91)

/usr/local/lib/python2.7/site-packages/fitz/fitz.pyc in __init__(self, *args)
   1696         __init__(self, doc, xref) -> Pixmap
   1697         """
-> 1698         this = _fitz.new_Pixmap(*args)
   1699         try:
   1700             self.this.append(this)

RuntimeError: xref not an image
```

**Comment by asshatter** [comment]
Yep, the xref is correct and I'm sorry I can't send the file due to privacy policy of my company.

**Comment by JorjMcKie** [comment]
That's unfortunate. Maybe your company can reconsider this, if you use my private e-mail and not the public GitHub channel ... you could also create a sub-PDF containing just the problem page.

But as I said: the PDF has some problem then. I still do not understand, why the HTML extraction should work in this situation though: both functions use the same base64 encoder ...

**Comment by JorjMcKie** [comment]
Meanwhile I have taken another look, more closely comparing JSON and HTML output.
I indeed found out that they not always do exactly the same thing with respect to base64 encoding. I am correcting this and will publish a fix today as version 1.12.6.

Please try it out and close this issue if you find it resolved.

As general hints for using getText JSON-output:
* Even if a block `b` is of type "image", `b["image"]` may be `None`. This occurs when the underlying MuPDF library encounters an image format it does not know about. So, this condition must be checked.
* Using base64 back-conversions of `b["image"]` should always be enclosed in `try ... except`. Neither MuPDF nor PIL or other packages are always happy with what we feed them.

**Comment by asshatter** [comment]
Hey, I just fixed your fitz_wrap.c and it worked for me. I'm forking you repo and and I'll make a pull request and you can review it and use it for next versions.

**Comment by JorjMcKie** [comment]
Hi, I am eager to see your PR dropping in!
If you however find the published fix in v1.12.6 sufficient, please let me know and close the issue.

**Comment by asshatter** [comment]
Hey, I actually made the changes in my local repo and installed it. But v 1.12.6 is sufficient. Tested it out on every file. Looks good. I'm closing the issue. Thank you.
